control-plane/pkg/kafka/clientpool/client.go

```diff
@@ -0,0 +1,238 @@
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) Leader(topic string, partitionID int32) (*sarama.Broker, error) {
+	x, err := c.client.Leader(topic, partitionID)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) LeaderAndEpoch(topic string, partitionID int32) (*sarama.Broker, int32, error) {
+	x, y, err := c.client.LeaderAndEpoch(topic, partitionID)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, y, err
+}
+
+func (c *client) Replicas(topic string, partitionID int32) ([]int32, error) {
+	x, err := c.client.Replicas(topic, partitionID)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) InSyncReplicas(topic string, partitionID int32) ([]int32, error) {
+	x, err := c.client.InSyncReplicas(topic, partitionID)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) OfflineReplicas(topic string, partitionID int32) ([]int32, error) {
+	x, err := c.client.OfflineReplicas(topic, partitionID)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) RefreshBrokers(addrs []string) error {
+	err := c.client.RefreshBrokers(addrs)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return err
+}
+
+func (c *client) RefreshMetadata(topics ...string) error {
+	err := c.client.RefreshMetadata(topics...)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return err
+}
+
+func (c *client) GetOffset(topic string, partitionID int32, time int64) (int64, error) {
+	x, err := c.client.GetOffset(topic, partitionID, time)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) Coordinator(consumerGroup string) (*sarama.Broker, error) {
+	x, err := c.client.Coordinator(consumerGroup)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) RefreshCoordinator(consumerGroup string) error {
+	err := c.client.RefreshCoordinator(consumerGroup)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return err
+}
+
+func (c *client) TransactionCoordinator(transactionID string) (*sarama.Broker, error) {
+	x, err := c.client.TransactionCoordinator(transactionID)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) RefreshTransactionCoordinator(transationID string) error {
+	err := c.client.RefreshTransactionCoordinator(transationID)
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return err
+}
+
+func (c *client) InitProducerID() (*sarama.InitProducerIDResponse, error) {
+	x, err := c.client.InitProducerID()
+	if c.isFatalError(err) {
+		c.onFatalError(err)
+	}
+	return x, err
+}
+
+func (c *client) LeastLoadedBroker() *sarama.Broker {
+	return c.client.LeastLoadedBroker()
+}
+
+func (c *client) Close() error {
+	// guard against panic, this makes it easier to defer client.Close() on the caller side
+	if c == nil {
+		return nil
+	}
```

Comment:
When does this case happen?

Reply:
Oh I've fixed this now, we can probably remove it. Basically, if you do something like:
```go
client, err := r.GetKafkaClient(...)
defer client.Close()
if err != nil {
   ...
}
```

Here, if `err` is not `nil`, the deferred `client.Close()` method panics. 

I fixed all the instances where this was happening, so I don't think we actually need this now